api/operator/v1beta1/vmalert_types.go

```diff
@@ -186,10 +186,10 @@ type VMAlertSpec struct {
 	// +optional
 	Notifiers []VMAlertNotifierSpec `json:"notifiers,omitempty"`
 
-	// NotifierConfigRef reference for secret with notifier configuration for vmalert
+	// NotifierConfigRef reference for configmap or secret with notifier configuration for vmalert
 	// only one of notifier options could be chosen: notifierConfigRef or notifiers +  notifier
 	// +optional
-	NotifierConfigRef *v1.SecretKeySelector `json:"notifierConfigRef,omitempty"`
+	NotifierConfigRef *SecretOrConfigMap `json:"notifierConfigRef,omitempty"`
```

Comment:
It's a serious breaking change without easy migration path. 

Since notifierConfig is widely used I cannot recommend to do that.

Reply:
I think it's better to add inlineNotifierConfig, which allows to post insecure configuration directly to `VMAlert` definition.